File: src/main/java/com/iminling/core/annotation/Validate.java
package com.iminling.core.annotation;

import com.iminling.core.validation.group.ValidateType;

import java.lang.annotation.*;

@Target({ElementType.TYPE, ElementType.METHOD, ElementType.PARAMETER})
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface Validate {

    Class<? extends ValidateType>[] value() default {};

}

File: iminling-core/src/main/kotlin/com/iminling/core/annotation/Validate.kt
package com.iminling.core.annotation

import com.iminling.core.validation.group.ValidateType
import kotlin.reflect.KClass

/**
 * @author  yslao@outlook.com
 * @since  2021/11/28
 */
@Target(AnnotationTarget.TYPE, AnnotationTarget.FUNCTION, AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class Validate(val value: Array<KClass<out ValidateType>> = [])
